### ISSUE TITLE
datacite: add file formats file sizes to datacite export format

### ISSUE THREAD

**Issue by NiekVanWettere** [zenodo/zenodo#2208]
## Is your feature request related to a problem? Please describe.
It strikes me that Zenodo metadata does not automatically contain information for "sizes" and "formats" of files. Isn't this something that, in principle, could be automatically derived from the uploaded files? See for example https://coptr.digipres.org/index.php/File_Format_Identification for identification of file formats via tools. According to the Zenodo helpdesk, the Zenodo API does extract the file metadata, but does not add it to the record metadata.

## Describe the solution you'd like
Automatically add relevant file metadata to the record metadata, so that it is available via metadata hubs such as OpenAire, DataCite etc. In particular, file formats are increasingly part of the assessment of the openness of data, so if we can improve the metadata completeness in this regard, that would be a great help.

**Comment by lnielsen** [comment]
What in particular are you looking for?

The ``/api/records/:id``endpoint have size and file type: Example

```python
{
  "conceptdoi": "10.5281/zenodo.3931948", 
  "conceptrecid": "3931948", 
  "created": "2021-09-02T09:09:09.437983+00:00", 
  "doi": "10.5281/zenodo.5378074", 
  "files": [
    {
      "bucket": "1ea33e12-8212-44df-abd9-1710b1a68e8c", 
      "checksum": "md5:f970040f1ed2181a7b7a001c55211300", 
      "key": "flightlist_20190101_20190131.csv.gz", 
      "links": {
        "self": "https://zenodo.org/api/files/1ea33e12-8212-44df-abd9-1710b1a68e8c/flightlist_20190101_20190131.csv.gz"
      }, 
      "size": 184000288, 
      "type": "gz"
    }, 
```

And if you follow the bucket link from the record metadata you get a mime-type (e.g https://zenodo.org/api/files/1ea33e12-8212-44df-abd9-1710b1a68e8c):

```python
...
    "contents": [{
        "mimetype": "application/gzip",
        "updated": "2021-09-02T09:09:08.764717+00:00",
        "links": {
            "self": "https://zenodo.org/api/files/1ea33e12-8212-44df-abd9-1710b1a68e8c/flightlist_20190101_20190131.csv.gz",
            "version": "https://zenodo.org/api/files/1ea33e12-8212-44df-abd9-1710b1a68e8c/flightlist_20190101_20190131.csv.gz?versionId=2590fbd5-cc22-42a3-a3ab-f59620554562",
            "uploads": "https://zenodo.org/api/files/1ea33e12-8212-44df-abd9-1710b1a68e8c/flightlist_20190101_20190131.csv.gz?uploads"
        },
        "is_head": true,
        "created": "2021-09-02T09:09:08.757574+00:00",
        "checksum": "md5:f970040f1ed2181a7b7a001c55211300",
        "version_id": "2590fbd5-cc22-42a3-a3ab-f59620554562",
        "delete_marker": false,
        "key": "flightlist_20190101_20190131.csv.gz",
        "size": 184000288
```

**Comment by NiekVanWettere** [comment]
Yes, the information is clearly available at Zenodo. But, for example, concerning the dataset you mention (10.5281/zenodo.3931948), its corresponding DataCite metadata record (https://api.datacite.org/dois/application/vnd.datacite.datacite+json/10.5281/zenodo.3931948) does not contain metadata for formats and sizes metadata. I have the impression that this is the case for most of the datasets published at Zenodo?

**Comment by lnielsen** [comment]
Got it, so it's in the DataCite export format you'd like to have it available in as well and not only in the JSON.

**Comment by NiekVanWettere** [comment]
Basically, the possibility to easily retrieve this information when connecting with European metadata hubs such as DataCite, without having to connect directly with Zenodo.

**Comment by lnielsen** [comment]
Note that DataCite does not allow linking to the data files themselves, and also the sizes/format fields of DataCite metadata schema are free-text fields so the information we put in there is really only useful for display purposes and not even for search. Having a standardized way to do this in the DataCite  would be useful in many other contexts as well, e.g. for harvesting/mirroring content.
